### DISCUSSION TITLE
Can't follow the route

### DISCUSSION THREAD

**Discussion by antonnurapkin** [flet-dev/flet#1832] [Q&A]
After authorization I want to go to home page, but it doesnt happen, I again get to the authorization page.

def _view_(page:Page):

    def login(e):
        e.page.launch_url(page.url+'/home')

    view = View('',controls=[TextButton('Login',
                                            on_click=login
                                            )])
    
    return view

**Comment by antonnurapkin** [comment]
Also tried this

def login(e):
        webbrowser.open_new_tab(page.url+'/home')

**Comment by gbrandt** [comment]
The proper way is to call `page.go( route )` and then to handle the change in page on the `on_route()` handler.  See here: https://flet.dev/docs/guides/python/navigation-and-routing

**Comment by antonnurapkin** [comment]
@gbrandt 
I want to create a new browser tab so that I have two tabs: a page with the old route and a page with the new one.
In your method it follows the route in current tab(page)

**Comment by gbrandt** [comment]
Ahh, my misunderstanding.  On Safari, it is currently impossible to open a new tab, you will briefly see 'pop-up blocked' in the window if you try.  On MacOS` launch_url( url, "_self")` will replace the current tab, on others `launch_url( url, "_blank")` will open a new tab. See https://github.com/flet-dev/flet/issues/1105 for more info.

**Comment by FeodorFitsner** [comment]
@antonnurapkin if it's Safari and popup-blocker then I suggest to close this issue in favour of #1105?

**Comment by antonnurapkin** [comment]
@FeodorFitsner to my mind it's not the same issue
I want to use the route as a URL and follow it while displaying a new page on a new tab of my browser(Yandex) .

Is it real?
